### PR TITLE
 [Symfony 6] Third solution to fix resource controllers

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | yes (for Symfony 6)
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | partially #400
| License         | MIT

### CODE REVIEW

File: src/Bundle/DependencyInjection/Compiler/TwigPass.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ResourceBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final class TwigPass implements CompilerPassInterface
```
Comment (lchrusciel) [inline src/Bundle/DependencyInjection/Compiler/TwigPass.php:22]: Let's add note, that this class will be removed in Resource 2.0

**Review by lchrusciel — APPROVED** [review]

File: src/Bundle/DependencyInjection/Compiler/TwigPass.php (hunk 1 above)
Comment (loic425) [reply src/Bundle/DependencyInjection/Compiler/TwigPass.php:22]: done

Comment (GSadee) [reply src/Bundle/DependencyInjection/Compiler/TwigPass.php:22]: Shouldn't we deprecate this class and inform about removing setting twig as public service in the UPGRADE file?

Comment (loic425) [reply src/Bundle/DependencyInjection/Compiler/TwigPass.php:22]: I think `deprecate it` should be landed with Sylius resource controller split (next release). Cause we need this usage for this release and can not remove using it.

**Comment by lchrusciel** [comment]
Thank you, Loïc! :1st_place_medal:

**Pull request merged**